File: 0x08-python-more_classes/4-rectangle.py
#!/usr/bin/python3
"""Class Rectangle"""


class Rectangle:
    """Define a rectangle"""

    def __init__(self, width=0, height=0):
        """ Start a new rectangle
        args:
        width: init
        height: init

        """
        self.width = width
        self.height = height

    @property
    def width(self):
        """Get width"""
        return self.__width

    @width.setter
    def width(self, value):
        """Set the width value"""
        if not isinstance(value, int):
            raise TypeError("width must be an integer")
        if value < 0:
            raise ValueError("width must be >= 0")
        self.__width = value

    @property
    def height(self):
        """Get height"""
        return self.__height

    @height.setter
    def height(self, value):
        """Set value"""
        if not isinstance(value, int):
            raise TypeError("height must be an integer")
        if value < 0:
            raise ValueError("height must be >= 0")
        self.__height = value

    def area(self):
        """Return the are of the rectangle"""
        return (self.__width * self.__height)

    def perimeter(self):
        """Return the perimeter of the rectangle"""
        if self.__width == 0 or self.__height == 0:
            return (0)
        return ((self.__width * 2) + (self.__height * 2))

    def __str__(self):
        """Return the printable representation of the rectangle
        Represent the rectangle with #
        """
        if self.__width == 0 or self.__height == 0:
            return ("")

        a = []
        for x in range(self.__height):
            [a.append('#') for y in range(self.__width)]
            if x != self.__height - 1:
                a.append("\n")
        return ("".join(a))

    def __repr__(self):
        """Return the string representation of the rectangle"""
        a = "Rectangle(" + str(self.__width)
        a += ", " + str(self.__height) + ")"
        return (a)

File: 0x13-javascript_objects_scopes_closures/4-rectangle.js
#!/usr/bin/node
class Rectangle {
  constructor (w, h) {
    if ((w > 0) && (h > 0)) {
      this.width = w;
      this.height = h;
    }
  }

  print () {
    for (let z1 = 0; z1 < this.height; z1++) {
      let a = '';
      for (let z2 = 0; z2 < this.width; z2++) {
        a += 'X';
      }
      console.log(a);
    }
  }

  rotate () {
    const aux = this.width;
    this.width = this.height;
    this.height = aux;
  }

  double () {
    this.width *= 2;
    this.height *= 2;
  }
}

module.exports = Rectangle;
